### PR TITLE
Add fish completion support

### PR DESCRIPTION
**Pull request by davewongillies**
Initial implementation of fish completion for chezmoi. Currently waiting on https://github.com/spf13/cobra/pull/754 to get merged, then I'll clean up `go.mod`.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -37,3 +36,7 @@ require (
 	golang.org/x/sys v0.0.0-20190310054646-10058d7d4faa // indirect
 	gopkg.in/yaml.v2 v2.2.2
 )
+
+// https://github.com/0robustus1/cobra/tree/350-add_fish_completion_support
+// https://github.com/spf13/cobra/pull/754
+replace github.com/spf13/cobra => ../cobra
```
Comment (twpayne) [inline go.mod]: Note that this replacement will only work on your local machine as it's using a relative path.

Note also that replace statements are ignored by `go get` (see https://github.com/golang/go/issues/31173 and #289).

**Comment by twpayne** [comment]
Thanks very much for this! `github.com/spf13/cobra` can be slow to merge PRs, so let's give it a week or so. If https://github.com/spf13/cobra/pull/754 doesn't get merged soon I'm happy to create a temporary fork with the fish completion included for use in chezmoi - at least until fish completion gets merged upstream.

File: go.mod (hunk 1 above)
Comment (davewongillies) [reply go.mod]: @twpayne oh yeah sorry, my comment should've been a bit clearer that this is temporary while I wait for the upstream PR to be merged. Its also why I raised this a draft PR.

Comment (twpayne) [reply go.mod]: Cool, yup got that this is a draft. Thanks for opening it so that people know that the work is in progress.

Comment (davewongillies) [reply go.mod]: I've updated `go.mod` to be useable outside of my machine 🙂

File: cmd/completion.go [hunk 2]
```diff
@@ -13,7 +13,7 @@ var completionCmd = &cobra.Command{
 	Short:     "Output shell completion code for the specified shell (bash or zsh)",
```
Comment (twpayne) [inline cmd/completion.go]: Please add `fish` here.

**Comment by twpayne** [comment]
It doesn't look like https://github.com/spf13/cobra/pull/754 is going to be merged any time soon, but I'd still like to see this in chezmoi. If you agree, could you mark this PR as "Ready for review". I'm happy to use @0robustus1's fork while waiting for @spf13 or any Cobra maintainer to merge the upstream PR.

**Comment by davewongillies** [comment]
@twpayne sure let's do it 👍

**Comment by twpayne** [comment]
Many thanks! The CI might be a bit too strict at times.

**Comment by markstos** [comment]
Thanks for Fish completion!

**Pull request merged**